### DISCUSSION TITLE
does Stirling-PDF process asynchronously ?

### DISCUSSION THREAD

**Discussion by smehdux** [Stirling-Tools/Stirling-PDF#2002] [Q&A] [answered]
Hi,

We were looking to the sources and we've discovered that Stirling-PDF launches a unique **unoconv** process to convert files to PDF leveraging open office.

My question is: did anyone carry out a performance tests by converting dozens or hundreds of files ? How does it behave ?

I believe that the Stirling-pdf API doesn't support any kind of queue or a mechanism to deal with a huge number of requests because of - maybe - OpenOffice process which doesn't allow to launch multi instances of its process.

I look forward to read your reponses

**Comment by Frooodle** [comment, marked as answer]
We implement a max instance count for each operation depending on how heavy it is 
We plan to dynamically change this but currently it's static 

Due to the max instance count the job does 'queue' of sorts

**Comment by Frooodle** [comment]
https://github.com/Stirling-Tools/Stirling-PDF/blob/main/src%2Fmain%2Fjava%2Fstirling%2Fsoftware%2FSPDF%2Futils%2FProcessExecutor.java 
L48

**Reply by smehdux** [reply]
> https://github.com/Stirling-Tools/Stirling-PDF/blob/main/src%2Fmain%2Fjava%2Fstirling%2Fsoftware%2FSPDF%2Futils%2FProcessExecutor.java L48

I saw it is why I'm questioning your team, so if we build our version with let's say 3 instances, could the subsequent command (unoconv/LibreOffice) launches 3 Linux processes ? I'm not sure about launching too many LibreOffice instances.

**Reply by Frooodle** [reply]
If you set this count to 3 or for some reason run 3 spdf instances on same machine then yes it could run 3 libre office instances

**Reply by smehdux** [reply]
Thank you @Frooodle, I hope the next version will manage this limit in the config file.
should I create a feature request  or suggest it ?

**Reply by Frooodle** [reply]
Please create a feature request ticket!

**Comment by smehdux** [comment]
I've created it => #2003
